### PR TITLE
DOC: Fix references to `trimseq` in docstrings

### PR DESCRIPTION
**Pull request by ghost**
Fixing reference in `*trim`(e.g. [chebtrim](https://numpy.org/devdocs/reference/generated/numpy.polynomial.chebyshev.chebtrim.html) )

### CODE REVIEW

**Comment by rossbar** [comment]
I don't think this is necessary - `trimseq` and `trimcoef` are both defined in polyutils. The link appears be working in the [latest documentation](https://numpy.org/devdocs/reference/generated/numpy.polynomial.polyutils.trimcoef.html#numpy-polynomial-polyutils-trimcoef)

**Comment by charris** [comment]
Slightly off topic, the circleci doc builds have a ton of warnings due to backticks around function variables. I seem to recall @bsipocz saying that astropy was using double backticks to avoid this. Do we have a policy on this?

**Comment by mattip** [comment]
It seems to be only some signature names. For instance, there are 51 warnings about "`size`" when used like in [`truncate`](https://github.com/numpy/numpy/blob/0e5a362fdaae6aac2d82c0767cfff25eebc5856e/numpy/polynomial/_polybase.py#L759)'s docstring, which triggers a warning
```
WARNING: py:obj reference target not found: size
```

**Comment by rossbar** [comment]
> The docstring of "trimcoef" is also used as the docstring of "chebtrim", "hermtrim", etc., which cause broken links.

Thanks for the context @F3eQnxN3RriK . In that case it's not clear what the best course of action is. On the one hand, this proposed change will fix the ~6 sphinx-nitpick link warnings. OTOH it provides an explicit link to an interface that, while public, is not recommended for use directly (the polynomial package encourages the convenience classes as the primary interface).

Off the top of my head, I can't think of a solution that doesn't have minor downsides: we could template the docstrings for the convenience classes, but that can be tricky to figure out/maintain. I don't feel particularly strongly about it.

File: numpy/polynomial/polyutils.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def trimcoef(c, tol=0):
 
     See Also
     --------
-    trimseq
+    polynomial.polyutils.trimseq
```
Comment (mdhaber) [inline numpy/polynomial/polyutils.py:190]:
```suggestion
```

File: numpy/polynomial/polyutils.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def trimcoef(c, tol=0):
 
     See Also
     --------
```
Comment (mdhaber) [inline numpy/polynomial/polyutils.py:189]:
```suggestion
```

**Review by mdhaber — COMMENTED** [review]
Since neither the fixed nor broken links are perfectly desirable, I think the compromise is to remove the "See Also" until the `*trim` documentation is separate from the `trimcoef` documentation. Thanks @F3eQnxN3RriK!

**Pull request merged**